i2c/imx6ull/imx6ull-i2c.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX 6ULL I2C BUS driver
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Marek Bialowas
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <sys/msg.h>
+#include <posix/utils.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include <i2c.h>
+#include <i2c-msg.h>
+
+
+static int dev_ctl(msg_t *msg)
+{
+	i2c_devctl_t *i = (i2c_devctl_t *)msg->i.raw;
+
+	switch (i->type) {
+		case i2c_devctl_bus_write:
+			return i2c_busWrite(i->dev_addr, msg->i.data, msg->i.size);
+
+		case i2c_devctl_bus_read:
+			return i2c_busRead(i->dev_addr, msg->o.data, msg->o.size);
+
+		case i2c_devctl_reg_read:
+			return i2c_regRead(i->dev_addr, i->reg_addr, msg->o.data, msg->o.size);
+
+		default:
+			return -EINVAL;
+	}
+}
+
+
+static void thread(void *arg)
+{
+	uint32_t port = (uint32_t)arg;
+	msg_t msg;
+	unsigned long int rid;
+
+	for (;;) {
+		if (msgRecv(port, &msg, &rid) < 0) {
+			memset(&msg, 0, sizeof(msg));
+			msgRespond(port, &msg, rid);
```

Comment:
I think we shouldn't reply if `msgRecv` returns error

Reply:
no docs, copied from other available sources. What should be done?

Reply:
ok, I assume it's a fatal error so it will result in device driver exiting.

Reply:
Such case was never seen in nature (only if e.g. port hasn't been created). Commonly it we do just `continue`

Reply:
Ok, after looking at the kernel code I see that msgRecv() can exit with (on MMU archs):
 - `EINVAL` if the port is invalid or was closed (I don't know how a port can be closed, can it be done by other process?)
 - `EINTR` if we receive a signal, so my solution needs to be fixed (non-fatal error)
![Screenshot_20210923_130845](https://user-images.githubusercontent.com/3939983/134497214-2396d58e-f7d8-4d53-a975-df7027f26ebc.png)
 - `ENOMEM` if message copying to the target process would fail - this is a fatal error and we don't know what to do in case of system OOM, so exiting from non-critical driver might be a way to keep floating

I'll fix the the message loop to tread `EINTR` as non-fatal and exit on other errors.